### PR TITLE
[improve][broker] Make some methods of ClusterBase pure async. 

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

See PIP #14365  and change tracker #15043. 

### Modifications

- Make `ClusterBase` `deleteNamespaceIsolationPolicy` / `setFailureDomain` / `getFailureDomains` to pure async.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

`AdminTest#clusters` already cover this change.

### Documentation

- [x] `no-need-doc` 
(Please explain why)

### CODE REVIEW

**Review by coderzc — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java [hunk 1]
```diff
@@ -686,46 +687,38 @@ private CompletableFuture<Void> filterAndUnloadMatchedNamespaceAsync(NamespaceIs
...
+                .thenCompose(__ -> validatePoliciesReadOnlyAccessAsync())
+                .thenCompose(__ -> namespaceIsolationPolicies().getIsolationDataPoliciesAsync(cluster))
+                .thenCompose(nsIsolationPoliciesOpt -> nsIsolationPoliciesOpt.map(CompletableFuture::completedFuture)
+                        .orElseGet(() -> namespaceIsolationPolicies()
+                                .setIsolationDataWithCreateAsync(cluster, (p) -> Collections.emptyMap())
+                                .thenApply(__ -> new NamespaceIsolationPolicies())))
+                .thenCompose(policies -> {
+                    policies.deletePolicy(policyName);
+                    return namespaceIsolationPolicies().setIsolationDataAsync(cluster, old -> policies.getPolicies());
+                }).thenAccept(__ -> asyncResponse.resume(Response.ok().build()))
```
Comment (shibd) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java:707]:
```suggestion
                }).thenAccept(__ -> asyncResponse.resume(Response.noContent().build()))
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java [hunk 2]
```diff
@@ -742,38 +735,37 @@ public void deleteNamespaceIsolationPolicy(
...
+    ) {
+        validateSuperUserAccessAsync()
+                .thenCompose(__ -> validateClusterExistAsync(cluster, PRECONDITION_FAILED))
+                .thenCompose(__ -> validateBrokerExistsInOtherDomain(cluster, domainName, domain))
+                .thenCompose(__ -> clusterResources().getFailureDomainResources()
+                        .setFailureDomainWithCreateAsync(cluster, domainName, old -> domain))
+                .thenAccept(__ -> {
+                    log.info("[{}] Successful set failure domain {} for cluster {}",
+                            clientAppId(), domainName, cluster);
+                    asyncResponse.resume(Response.ok().build());
```
Comment (shibd) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java:753]:
```suggestion
                    asyncResponse.resume(Response.noContent().build());
```

**Review by shibd — CHANGES_REQUESTED** [review]
Need to keep the original response code

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java (hunk 2 above)
Comment (mattisonchao) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java]: Fixed.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java (hunk 1 above)
Comment (mattisonchao) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java]: Fixed.

**Review by shibd — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**